source/anti-patterns/language.rst

```diff
@@ -9,10 +9,46 @@ Nix syntax supports URLs as verbatim, so one can write ``https://example.com`` i
 
 There's was an `RFC 45 <https://github.com/NixOS/rfcs/pull/45>`_ accepted to deprecate verbatim URLS and provides
 a number of arguments this features does more harm than good.
+ 
 
+``rec { ... }`` expression
+--------------------------
 
-``with`` expression
--------------------
+``rec`` allows you to reference variables within an attribute set.
+
+A simple example:
+
+.. code:: nix
+
+  rec {
+    a = 1;
+    b = a + 2;
+  }
+
+evaluating to ``{ a = 1; b = 3; }``.
+
+``b`` refers to ``a`` as ``rec`` makes all keys available within.
+
+There are a couple of pitfals:
```

Comment:
I always wondered, is there a notable difference in performance with `rec` attrsets?

Reply:
Me too, but that's out of scope here :)

Reply:
Well, I think if use of `rec` has a significant impact on performance, it might be another reason to avoid it :)

Reply:
Right, going to add that once someone confirms it :)